### PR TITLE
set-ouput warnings during CI runs - latest from upstream

### PR DESCRIPTION
**Pull request by gvegusb**
The changes on this PR will make use of Environment Files instead of set-output commands. The set-output command is deprecated and will be disabled soon. these changes are necessary to prevent run fails in the future.

Create Task : https://tbct-jira.valiantys.net/browse/ATL-16128
Review Task: https://tbct-jira.valiantys.net/browse/ATL-16130

### CODE REVIEW

**Comment by dmacusb2** [comment]
@gvegusb you'll have to resolve the merge conflicts

File: action.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ inputs:
   head-ref:
     description: 'ref of the head commit to compare (mainly for testing the action)'
 runs:
-  using: 'node12'
+  using: 'node16'
```
Comment (dmacusb2) [inline action.yml:20]: Can you bump this to 'node20'

Currently this is still generating:

```
[build](https://github.com/TBCTSystems/bct-common-device-standard-messages/actions/runs/7761300754/job/21169424756)
Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/checkout@v2, ./.github/actions/actions-changed-file-filter. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.
```

**Comment by gvegusb** [comment]
changed from using node 16 to node 20

File: action.yml (hunk 1 above)
Comment (gvegusb) [reply action.yml]: changed from using node 16 to node 20

**Review by dmacusb2 — APPROVED** [review]

**Pull request merged**